extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -615,7 +615,9 @@ public void updatePartialKeyStatisticsInformation(int stageNumber, int workerNum
           queryKernel.addPartialKeyStatisticsForStageAndWorker(stageId, workerNumber, partialKeyStatisticsInformation);
 
           if (queryKernel.getStagePhase(stageId).equals(ControllerStagePhase.MERGING_STATISTICS)) {
-            List<String> workerTaskIds = workerTaskLauncher.getTaskList();
+            // we only need tasks which are active for this stage.
+            List<String> workerTaskIds = workerTaskLauncher.getTaskList()
```

Comment:
Yes, we can be sure that getTaskList() is active tasks only. 
It's kind of a coincidence that the moment you commented, I changed this logic to a less brittle one.

Reply:
Thanks for the fix, @cryptoe , the new method does seem better. I would also suggest just doing the filtering of `workerTaskIds` in the controller itself rather than passing an extra argument to the `WorkerSketchFetcher` just to filter the task ids later. Then the logic in the `WorkerSketchFetcher` wouldn't have to change and it wouldn't have to be aware of the worker indexes (which seems like an impl detail of the controller).
Hope this makes sense.

Reply:
Sounds good.